include/mls/messages.h

```diff
@@ -455,12 +480,52 @@ struct Sender
   TLS_SERIALIZABLE(sender_type, sender)
 };
 
+struct MLSPlaintextTBS
+{
+  const var::variant<GroupContext>& context;
```

Comment:
This is going to produce non-spec-compliant serialized objects: (a) it will always serialize the GroupContext [1], and (b) it will serialize an extra octet at the front for the SenderType [2].  Instead, we need to (a) only serialize the GroupContext when `sender_type == member` and (b) then just serialize the GroupContext.

It's not totally clear how to do this most elegantly, but the best solution that comes to my mind is:
* Extend the `tls::variant` trait to have a template parameter (say `implicit_selector`) that says whether the header is serialized
    * On encode, `if (!implicit_selector) { /* serialize selector */ }`
    * On decode, `if (implicit_selector) { /* throw; this is not possible */ }`
* Change this field type to `var::variant<GroupContext, struct{}>`
* Set the appropriate value in `MLSPlaintext::to_be_signed()` according to `sender.type`.

On a different note: I'm surprised this is passing tests!  Since we support external joins, this seems to imply that an external joiner has enough state to compute `context` for the previous group -- or the tests are broken.  If the former, we should update the spec so that external commits are signed with `context` as well.

[1] This might not be clear from the spec, but the intent of that `select` is that for cases other than `member`, nothing will be serialized, i.e., there's an unstated branch `default: struct{};`.  We should probably add that to clarify.

[2] `var::variant` serialization optimizes for the common case where you precede the variant with a selector, as for `Proposal` and `MLSPlaintext.content` (and a few places in TLS).  It requires that the selector be there, though.

Reply:
Re [1]: Yes, this was not clear from the spec at all, though in retrospect it makes sense. Since other variants do not have such a default, it seems like this should be explicit in the spec?